output-gazetteer.cpp

```diff
@@ -692,22 +692,20 @@ int output_gazetteer_t::process_way(osmid_t id, osmid_t *ndv, int ndc,
 
     /* Are we interested in this item? */
     if (places.has_data()) {
-        struct osmNode *nodev;
         int nodec;
 
+        assert(ndc > 0 && "no nodes to proccess");
```

Comment:
I added the assertion, because otherwise your old code would have done a malloc(0), which quoting the docs
> returns either NULL, or a unique pointer value that can later be successfully passed to free()

only passing it to free is guaranteed to work, but not e.g. dereferencing it, even in cases where it is _not_ a null pointer.

It's the same with &nodev[0]; if ndc == 0, &nodev[0] is illegal (this could even be simplified in C++11 with nodev.data() which returns nullptr in this case).

If ndc == 0 can indeed happen, the code has to be rewritten with this in mind.